hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -337,9 +342,24 @@ private com.hederahashgraph.api.proto.java.NftTransfer findNftTransferInsideBody
     }
 
     private void insertFungibleTokenTransfers(
-            long consensusTimestamp, TransactionBody body, boolean isTokenDissociate,
-            TokenID tokenId, EntityId entityTokenId, EntityId payerAccountId, List<AccountAmount> tokenTransfers) {
+            RecordItem recordItem, TokenID tokenId, EntityId entityTokenId,
+            EntityId payerAccountId, List<AccountAmount> tokenTransfers) {
+        long consensusTimestamp = recordItem.getConsensusTimestamp();
+        TransactionBody body = recordItem.getTransactionBody();
+        boolean isTokenDissociate = body.hasTokenDissociate();
         boolean isDeletedTokenDissociate = isTokenDissociate && tokenTransfers.size() == 1;
+        int tokenTransferCount = tokenTransfers.size();
+        
+        // If the last accountAmount in the transferList is with amount below zero it's a burn or wipe
+        boolean isWipeOrBurn = tokenTransfers.get(0).getAmount() < 0 && tokenTransfers.get(tokenTransferCount-1).getAmount() < 0;
+        // If the first accountAmount in the transferList is with amount above zero it's a mint
+        boolean isMint = tokenTransfers.get(0).getAmount() > 0 && tokenTransfers.get(tokenTransferCount-1).getAmount() > 0;
+        boolean isNotMultiTransfer = true;
+        if (tokenTransferCount > 1 && !isMint && !isWipeOrBurn) {
+            // If we have more than one accountAmounts, first one is always a negative amount and the second is positive
+            isNotMultiTransfer = tokenTransfers.get(0).getAmount() < 0 && tokenTransfers.get(1).getAmount() > 0;
+        }
```

Comment:
I'm not sure that you can rely upon transfer list order. Even if you can now, that may not be the case in the future and is brittle to use. Why can't you just do:
```suggestion
        boolean isNotMultiTransfer = tokenTransfers.size() <= 2 && !isMint && !isWipeOrBurn;
```

Or even just:

```suggestion
        boolean isSingleTransfer = tokenTransfers.size() == 2;
```

Reply:
This one is a little trickier. If we do it like you're proposing, we are losing support for one sender with multiple receivers. I agree that's not good to rely on transfer order.

Reply:
Please let's make sure we have test coverage for each of these scenarios. What about counting positive transfers up front, then only generate logs if the positive count is one and not wipe or burn?

Reply:
Not sure it would work, because to be a single transfer it needs one negative and one or more positive. So checking for only positive counts is not covering this. I'm maybe leaning towards your suggestion for checking, if transfer list count == 2. This way we cover single transfers with one receiver and one sender. And for the other two cases - one sender -> multiple receivers and multiple senders -> multiple receivers - we are not supporting them for now, until we find a proper way. What do you think ?

Reply:
Probably safest for now.

Reply:
Okay, let's settle for single transfer -> one sender, one receiver. After this PR we can open a issue to capture multi transfers (research if there is a way currently to do them correctly).